compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -209,7 +209,7 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
 
     tcx.ensure().check_unused_traits(());
 
-    if let Some(reported) = tcx.dcx().has_errors() { Err(reported) } else { Ok(()) }
+    Ok(())
```

Comment:
this is the motivation of this PR. removing another `has_errors` that affects compilation progress

Reply:
:+1: